File: backend/app/Models/Payment.php
<?php

namespace App\Models;

use App\Enums\PaymentStatus;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Payment extends Model
{
    protected $fillable = [
        'dining_session_id',
        'status',
        'amount_due',
        'amount_paid',
        'paid_at',
        'marked_by_staff_id',
    ];

    protected function casts(): array
    {
        return [
            'status' => PaymentStatus::class,
            'amount_due' => 'decimal:2',
            'amount_paid' => 'decimal:2',
            'paid_at' => 'datetime',
        ];
    }

    public function diningSession(): BelongsTo
    {
        return $this->belongsTo(DiningSession::class);
    }

    public function markedByStaff(): BelongsTo
    {
        return $this->belongsTo(User::class, 'marked_by_staff_id');
    }
}

File: lib/features/payments/domain/entities/payment.dart
import 'package:equatable/equatable.dart';

import '../../../shared/domain/enums.dart';

class Payment extends Equatable {
  const Payment({
    required this.id,
    required this.sessionId,
    required this.status,
    required this.amountDue,
    this.amountPaid,
    this.paidAt,
    this.markedByStaffId,
  });

  final String id;
  final String sessionId;
  final PaymentStatus status;
  final double amountDue;
  final double? amountPaid;
  final DateTime? paidAt;
  final String? markedByStaffId;

  bool get isPaid => status == PaymentStatus.paid;

  @override
  List<Object?> get props => [
        id,
        sessionId,
        status,
        amountDue,
        amountPaid,
        paidAt,
        markedByStaffId,
      ];
}
